### PR TITLE
Rejection email

### PR DESCRIPTION
**Pull request by leshyabracaglia**
Linked to hacknyu twitter, but if we set up our own twitter feed we can change link to that

### CODE REVIEW

File: functions/src/RejectedEmail.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            <div style={{ padding: "40px", fontSize: "1.1em" }}>
+                <p>
+                    Dear {name ? ` ${name}` : ""},
+                </p>
+                <p>
+                     Thank you for submitting you application to HackNYU! 
+                </p>
+                <p>
+                  Due to limited space, we regret to inform you that we are
+                  not able to accept you as a participant this year.</p>
```
Comment (themichaelyang) [inline functions/src/RejectedEmail.tsx]: Let's make sure indentation is consistent throughout

File: functions/src/RejectedEmail.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                  <p>HackNYU takes place simultaneously in New York, Shanghai and Abu Dhabi over 48
+                  hours. This year, we received over 2000 applications, making us the top university hackathon in New York!
```
Comment (themichaelyang) [inline functions/src/RejectedEmail.tsx]: maybe switch `top` to `biggest`?

File: functions/src/RejectedEmail.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                <p>
+                    Dear {name ? ` ${name}` : ""},
+                </p>
+                <p>
+                     Thank you for submitting you application to HackNYU! 
+                </p>
+                <p>
+                  Due to limited space, we regret to inform you that we are
+                  not able to accept you as a participant this year.</p>
+                  <p>HackNYU takes place simultaneously in New York, Shanghai and Abu Dhabi over 48
```
Comment (themichaelyang) [inline functions/src/RejectedEmail.tsx]: This sentence feels out of place

File: functions/src/RejectedEmail.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                  hours. This year, we received over 2000 applications, making us the top university hackathon in New York!  
+              </p>
+                  <p>You can check up on HackNYU 2019 from our twitter feed
+                          {" "}<a href="https://twitter.com/hacknyu">
+                                  here
+                              </a>.
+                  </p>
+                  <p>We hope to see your application next year!</p>
+                  <p>All the Best,</p>
```
Comment (themichaelyang) [inline functions/src/RejectedEmail.tsx]: lowercase "best"

File: functions/src/RejectedEmail.tsx (hunk 2 above)
Comment (leshyabracaglia) [reply functions/src/RejectedEmail.tsx]: I though it maybe gave a more positive explanation. Should I just delete that part? It is a bit awkward

File: functions/src/index.ts [hunk 5]
```diff
@@ -63,6 +65,55 @@ export const sendAcceptanceEmail = functions.firestore
...
+        text: "Please confirm your spot at HackNYU 2019",
+        html
+      };
+      return sgMail.send(msg, true);
+    })
+    .then(() => console.log("SENT"))
+    .catch(err => console.error(err.response.body));
+};
+
+sendReminderEmail();
```
Comment (themichaelyang) [inline functions/src/index.ts]: wouldn't this send the email every deploy?

File: functions/src/index.ts [hunk 6]
```diff
@@ -190,14 +190,38 @@ const sendRejectionEmail = () => {
...
+  const auth = admin.auth();
+  const admittedUsers = {};
+
+  db
+    .collection("admitted")
+    .get()
+    .then(snapshot => {
+      snapshot.forEach(doc => {
+        const data = doc.data();
+        admittedUsers[doc.id] = { 'firstName': data.firstName, 'confirmed': !data.confirmData };
```
Comment (stefancherubin) [inline functions/src/index.ts]: Hmm, I'm a little confused on the `!data.confirmData` part. Also a minor thing: we don't need their last name?

Comment (themichaelyang) [reply functions/src/index.ts]: I think it's fine if we just send it `Dear <firstName>`

File: functions/src/index.ts [hunk 7]
```diff
@@ -277,13 +227,39 @@ export const getRejectedUsers = () => {
...
   });
 };
 
-// sendRejectionEmail()
-//   .catch(err => console.error(err.toString()))
-//   .then(() => console.log("SENT!"));
+export const sendRejectionEmail = () => {
+  // const apiKey = functions.config().sendgrid.key;
+  // sgMail.setApiKey(apiKey);
+  const html = renderRejectedEmail("Billie Jean");
```
Comment (stefancherubin) [inline functions/src/index.ts]: Curious, is the "Billie Jean" just there as an easter egg?

Comment (themichaelyang) [reply functions/src/index.ts]: I think it was a test email Nicholas left in

File: functions/src/index.ts [hunk 8]
```diff
@@ -235,27 +235,33 @@ export const getRejectedUsersData = () => {
...
         to: data.email,
         from: "confirm@hacknyu.org",
         subject: "Update on Your Admission Status",
         text: "Update on your HackNYU Admission status",
         html: renderRejectedEmail(firstName)
       };
-      console.log(firstName + ': ' + data.email);
-      // return sgMail.send(msg, true);
-    }
+    // for (let uid of Object.keys(rejectedUsersData)) {
```
Comment (stefancherubin) [inline functions/src/index.ts]: I forgot to mention that I got the test email last night, so I think the email is ready to go(?), after these commented lines of code are gone

**Pull request merged**